### PR TITLE
add "main must return int" matcher

### PR DESCRIPTION
**Pull request by brianyu28**
Addressing #69, and `/review` inputs 1430, 1612, and 2053.

### CODE REVIEW

File: helpers/clang.py [hunk 1]
```diff
@@ -375,6 +375,23 @@ def help(lines):
 
     # $ clang foo.c
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:3:1: error: 'main' must return 'int'
+    # void main(void)
+    # ^~~~
+    # int
+    matches = re.search(r"^([^:]+):(\d+):\d+: error:", lines[0])
```
Comment (thctamm) [inline helpers/clang.py]: The actual error message ('main' must return 'int') seems to be missing

Comment (brianyu28) [inline helpers/clang.py]: Good catch, thanks.

File: helpers/clang.py [hunk 2]
```diff
@@ -12,18 +12,18 @@ def help(lines):
...
-    #                     ^~~~~ 
+    #                     ^~~~~
     matches = re.search(r"^([^:]+):(\d+):\d+: error: array subscript is not an integer", lines[0])
     if matches:
         array = var_extract(lines[1:3], left_aligned=False)
         index = tilde_extract(lines[1:3])
         if array and index:
             response = [
-                "Looks like you're trying to access an element of the array `{}` on line {} of `{}`, but your index `{}` is not of type `int`.".format(array, matches.group(2), matches.group(1), index)    
+                "Looks like you're trying to access an element of the array `{}` on line {} of `{}`, but your index `{}` is not of type `int`.".format(array, matches.group(2), matches.group(1), index)
```
Comment (dmalan) [inline helpers/clang.py]: Minor, but feels like index should be a parenthetical:

`but your index (\`{}\`) is`

File: helpers/clang.py [hunk 3]
```diff
@@ -375,6 +375,23 @@ def help(lines):
...
+    # ^~~~
+    # int
+    matches = re.search(r"^([^:]+):(\d+):\d+: error:", lines[0])
+    if matches:
+        response = [
+            "Your `main` function (declared on line {} of `{}`) must have a return type `int`.".format(matches.group(2), matches.group(1))
+        ]
+        if len(lines) >= 3:
+            cur_type = var_extract(lines[1:3])
+            response.append("Right now, it has return type `{}`.".format(cur_type))
```
Comment (dmalan) [inline helpers/clang.py]:
```
has a return type of `{}`
```

**Comment by dmalan** [comment]
:shipit:

**Pull request merged**